components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/API.java

```diff
@@ -28,6 +28,10 @@
 
     private APIIdentifier id;
 
+    // registry artifact id
+    // this id is provider's username independent
+    private String artifactId;
```

Comment:
Hmmm , might be inconsistent with other models may be we can call it UUID in all the other models as well.